reactos/drivers/usb/usbhub_new/pnp.c

```diff
@@ -1251,48 +1256,48 @@ USBH_FdoQueryBusRelations(IN PUSBHUB_FDO_EXTENSION HubExtension,
                     }
 
                     USBH_SyncResetPort(HubExtension, Port);
+
+                    ix++;
                 }
 
-                if (PortData->DeviceObject)
+                if (NT_SUCCESS(NtStatus))
                 {
-                    PdoExtension = PortData->DeviceObject->DeviceExtension;
-
-                    if (!(PdoExtension->PortPdoFlags & USBHUB_PDO_FLAG_PORT_LOW_SPEED) &&
-                        !(PdoExtension->PortPdoFlags & USBHUB_PDO_FLAG_PORT_HIGH_SPEED) &&
-                        !(HubExtension->HubFlags & USBHUB_FDO_FLAG_USB20_HUB))
+                    if (PortData->DeviceObject)
                     {
-                        DPRINT1("USBH_FdoQueryBusRelations: FIXME USBH_DeviceIs2xDualMode()\n");
+                        PdoExtension = PortData->DeviceObject->DeviceExtension;
 
-                        if (0)//USBH_DeviceIs2xDualMode(PdoExtension))
+                        if (!(PdoExtension->PortPdoFlags & USBHUB_PDO_FLAG_PORT_LOW_SPEED) &&
+                            !(PdoExtension->PortPdoFlags & USBHUB_PDO_FLAG_PORT_HIGH_SPEED) &&
+                            !(HubExtension->HubFlags & USBHUB_FDO_FLAG_USB20_HUB))
                         {
-                            PdoExtension->PortPdoFlags |= USBHUB_PDO_FLAG_HS_USB1_DUALMODE;
+                            DPRINT1("USBH_FdoQueryBusRelations: FIXME USBH_DeviceIs2xDualMode()\n");
+
+                            if (0)//USBH_DeviceIs2xDualMode(PdoExtension))
+                            {
+                                PdoExtension->PortPdoFlags |= USBHUB_PDO_FLAG_HS_USB1_DUALMODE;
+                            }
                         }
                     }
-                }
-
-          AddObject:
 
-                if (NT_SUCCESS(NtStatus))
-                {
                     ObReferenceObject(PortData->DeviceObject);
 
                     DeviceRelations->Objects[DeviceRelations->Count] = PortData->DeviceObject;
 
                     PortData->DeviceObject->Flags |= DO_POWER_PAGABLE;
                     PortData->DeviceObject->Flags &= ~DO_DEVICE_INITIALIZING;
 
-                    ++DeviceRelations->Count;
+                    DeviceRelations->Count++;
 
                     PortData->ConnectionStatus = DeviceConnected;
 
-                    goto NextPort;
+                    continue;
                 }
             }
         }
 
         PortData->ConnectionStatus = DeviceFailedEnumeration;
 
-        if ((USBH_SyncDisablePort(HubExtension, Port) & 0xC0000000) == 0xC0000000)
+        if (!NT_SUCCESS(USBH_SyncDisablePort(HubExtension, Port)))
```

Comment:
Technically this was an NT_ERROR check before. I don't know in how far that's important.